### ISSUE TITLE
some question about your paper

### ISSUE THREAD

**Issue by songhaohaohao** [ushaham/RBMpaper#1]
Dear Uri Shaham,
     I am so sorry to disturb you .Recently I have read your paper titled“A Deep Learning Approach to Unsupervised Ensemble Learning”.But I have some doubts about some parameters.
rbmInput.reg_type = 'l2';
rbmInput.weightPenalty = 1e-2; 
weightPenaltyOrg = rbmInput.weightPenalty;
rbmInput.epsilonw      = 5e-2;
rbmInput.epsilonvb     = 5e-2;
rbmInput.epsilonhb     = 5e-2;
rbmInput.decayLrAfter = 120;
[numClassifiers, numObs] = size(f);
orgData=f';

  In the above section
what's the meaning of 'reg_type' ,    'Org' ,   'epsilon' ,   'decayLrAfter',    'numObs' ,   'orgData'.
and why is   reg_type='12'    rbmInput.weightPenalty = 1e-2 and rbmInput.epsilonw = 5e-2

It's hard for me to understand what they mean.It's  great help for me  if you can answer my doubts 
thank you  very  much

```
Thanks in advance!
```

**Comment by ushaham** [comment]
hey, thx for reaching out.
- rbmInput.reg_type is type of weight penalty. l2 means that you
  penalize the l_2 norm of the weight vector.
- rbmInput.weightPenalty is the number by which you multiply the norm of
  the above weight vector. This term is added to the cost function.
- the epsilons are learning rates.
- orgData is your input matrix,
- numbs is the number of points in your dataset.

Let me know if you have any further questions, I'd be most happy to help.

On Thu, Oct 20, 2016 at 9:14 AM, songhaohaohao notifications@github.com
wrote:

> Dear Uri Shaham,
> I am so sorry to disturb you .Recently I have read your paper titled“A
> Deep Learning Approach to Unsupervised Ensemble Learning”.But I have some
> doubts about some parameters.
> rbmInput.reg_type = 'l2';
> rbmInput.weightPenalty = 1e-2;
> weightPenaltyOrg = rbmInput.weightPenalty;
> rbmInput.epsilonw = 5e-2;
> rbmInput.epsilonvb = 5e-2;
> rbmInput.epsilonhb = 5e-2;
> rbmInput.decayLrAfter = 120;
> [numClassifiers, numObs] = size(f);
> orgData=f';
> 
> In the above section
> what's the meaning of 'reg_type' , 'Org' , 'epsilon' , 'decayLrAfter',
> 'numObs' , 'orgData'.
> and why is reg_type='12' rbmInput.weightPenalty = 1e-2 and
> rbmInput.epsilonw = 5e-2
> 
> It's hard for me to understand what they mean.It's great help for me if
> you can answer my doubts
> thank you very much
> 
> Thanks in advance!
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ushaham_RBMpaper_issues_1&d=CwMFaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=YpOvWnVcv_J9_ZmCDwAz2zVOS348nEheo-bih9LENLc&m=0ZNEEgIzUBIimNPbP_FVo0aCS8JclJ80AJ6KkH82HLI&s=IB8UvILX1CsTghfBoXqoC6OHDWxNxROEJcmzzbXxLl8&e=,
> or mute the thread
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ACFAre39uT1J88rv7yh3Ppv9mmVoaVZ1ks5q12lNgaJpZM4KcHLa&d=CwMFaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=YpOvWnVcv_J9_ZmCDwAz2zVOS348nEheo-bih9LENLc&m=0ZNEEgIzUBIimNPbP_FVo0aCS8JclJ80AJ6KkH82HLI&s=RP72qGnI0EumpIW5zSrJWIk1FAHCnpKItJ5AK5EDum8&e=
> .

## 

Thx,
U.

**Comment by songhaohaohao** [comment]
hi, thank you for answering my question
in fact,   I don't know the whole framework of the main program, I  read the paper twice, also   read  program several times, but the steps of the program  don't  still  be  understood.   SVD  judges  the number of hidden layer and how does the parameters  train?I read  a  paper concerning   the classification of labeled data via RBM, now I want to know that  unsupervised learning methods  of  no labels
Thanks in advance!

**Comment by ushaham** [comment]
Hey,
The program gets a matrix of binary predictions of d classifiers on n
examples and outputs a 1xn binary prediction vector.
It works by training a single RBM at a time.
Wach RBM is trained at first using humber of hidden units = number of
visible units. Once trained, we use SVD to estimate the rank r of the
weight matrix, and re-train the RBM using r hidden units.
If r >1, we add an RBM on top of this one, and repeat the procedure.
This is the algorithm that is described in my ICML paper, and this is what
the matlab code is doing.
The last RBM in the stack has a single hidden unit, and this units gives
you the binary prediction of the true label.

I hope things are clearer now, let me know if you have any further
questions.

On Fri, Oct 21, 2016 at 4:54 AM, songhaohaohao notifications@github.com
wrote:

> hi, thank you for answering my question
> in fact, I don't know the whole framework of the main program, I read the
> paper twice, also read program several times, but the steps of the program
> don't still be understood. SVD judges the number of hidden layer and how
> does the parameters train?I read a paper concerning the classification of
> labeled data via RBM, now I want to know that unsupervised learning methods
> of no labels
> Thanks in advance!
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ushaham_RBMpaper_issues_1-23issuecomment-2D255329278&d=CwMFaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=YpOvWnVcv_J9_ZmCDwAz2zVOS348nEheo-bih9LENLc&m=2B983cDYeM9l19wzIRW0QW-QHfyTRfwL0freIZn0i6k&s=_FbHnuMXfruwmPQzmU9Dc2QMW2_U8wmkeQRlK837_-o&e=,
> or mute the thread
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ACFArWuwjmo73nCmaI40-5FCor190dACO1ks5q2H3PgaJpZM4KcHLa&d=CwMFaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=YpOvWnVcv_J9_ZmCDwAz2zVOS348nEheo-bih9LENLc&m=2B983cDYeM9l19wzIRW0QW-QHfyTRfwL0freIZn0i6k&s=CqTqwfKtKGQWgAaf8S5sTzp6tzXKRP6XQ63Sww-tQno&e=
> .

## 

Thx,
U.

**Comment by songhaohaohao** [comment]
hi, thank you for answering my question
I understand a lot more than before,  Because there are  data labels, the equivalent of rbm is supervised, 
data.trainData value is 100x15x90 double, which is what the 90 is the image of the order of what?
Thanks in advance!

**Comment by ushaham** [comment]
d=15 dimensions, 90 mini batches of 100 points each (so n=9000 training
points in total).

On Thu, Oct 27, 2016 at 5:40 AM, songhaohaohao notifications@github.com
wrote:

> hi, thank you for answering my question
> I understand a lot more than before, Because there are data labels, the
> equivalent of rbm is supervised,
> data.trainData value is 100x15x90 double, which is what the 90 is the
> image of the order of what?
> Thanks in advance!
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ushaham_RBMpaper_issues_1-23issuecomment-2D256595382&d=CwMFaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=YpOvWnVcv_J9_ZmCDwAz2zVOS348nEheo-bih9LENLc&m=iCdX42X7Qm0kMF1Gk8QpTMjN9VNdZjm33wH-tUm4CSo&s=0I_qOw1NmxSOiDQCHO_6EesptsiOgUjOPTLOBY7Gkqw&e=,
> or mute the thread
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ACFArfgUyXsNJYi-2DIxESubytv1YeK87Aks5q4HGKgaJpZM4KcHLa&d=CwMFaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=YpOvWnVcv_J9_ZmCDwAz2zVOS348nEheo-bih9LENLc&m=iCdX42X7Qm0kMF1Gk8QpTMjN9VNdZjm33wH-tUm4CSo&s=a11ipxCypQe2x4hxA19vID4WDf1Al0bSLu6CNoWUi38&e=
> .

## 

Thx,
U.
